src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.hmpps.whereabouts.services
+
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.whereabouts.controllers.VideoLinkAppointmentMigrationResponse
+import uk.gov.justice.digital.hmpps.whereabouts.model.HearingType
+import uk.gov.justice.digital.hmpps.whereabouts.model.PrisonAppointment
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkAppointment
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkBooking
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkAppointmentRepository
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkBookingRepository
+
+@Service
+class VideoLinkBookingMigrationService(
+
+  val videoLinkBookingRepository: VideoLinkBookingRepository,
+  val videoLinkAppointmentRepository: VideoLinkAppointmentRepository,
+  val prisonApiService: PrisonApiService,
+) {
+  fun migrateFromNomis(batchSize: Int): VideoLinkAppointmentMigrationResponse {
+
+    videoLinkBookingRepository.findByPrisonIdIsNull(PageRequest.of(0, batchSize)).stream().forEach { v -> updateVideoLink(v) }
+
+    return VideoLinkAppointmentMigrationResponse(
+      videoLinkAppointmentRepository.countByLocationIdIsNull(),
+      videoLinkBookingRepository.countByPrisonIdIsNull(),
+    )
+  }
+
+  fun updateVideoLink(videoLinkBooking: VideoLinkBooking) {
+    val mainAppointment = videoLinkBooking.appointments.get(HearingType.MAIN)
+    val perAppointment = videoLinkBooking.appointments.get(HearingType.PRE)
+    val postAppointment = videoLinkBooking.appointments.get(HearingType.POST)
+    if (mainAppointment == null) {
+      videoLinkAppointmentRepository.deleteAll(videoLinkBooking.appointments.values)
+      videoLinkBookingRepository.delete(videoLinkBooking)
+    } else {
+      val nomisMainAppointment = updateAppointment(mainAppointment)
+      if (nomisMainAppointment == null) {
+        videoLinkAppointmentRepository.deleteAll(videoLinkBooking.appointments.values)
```

Comment:
does cascade remove mean that this isn't necessary? can we not just delete the booking?

Reply:
simplified and confirmed by integration test